### PR TITLE
feat: update to support axios 1.x

### PR DESCRIPTION
**Pull request by ymor**
Updated the following packages too:
 - LRUCache
 - jest
 - ts-jest
 - prettier
 - typescript

### CODE REVIEW

**Review by simoneb — COMMENTED** [review]
Thanks a lot for this contribution, much appreciated! I never got around to doing the whole upgrade as it had too many moving pieces. There are a couple of ling warnings, would you mind fixing those as well?

**Comment by ymor** [comment]
Np, will try and take a look today.

Sent from Samsung Mobile on O2
Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ymor** [comment]
Looks like the cancel token is deprecated in favour of abort controller from `v0.22.0` . Reckon its worth moving it over too?

**Comment by simoneb** [comment]
Absolutely, can you do it within this PR?

**Comment by ymor** [comment]
Not sure if you get notified or not but that's in now too :)

File: package.json [hunk 1]
```diff
@@ -29,13 +29,13 @@
...
     "test": "tsd && jest --no-cache"
   },
   "dependencies": {
     "@babel/runtime": "7.22.11",
     "dequal": "2.0.3",
-    "lru-cache": "^8.0.0"
+    "lru-cache": "^10.0.1"
   },
   "peerDependencies": {
     "axios": ">=0.24.0",
```
Comment (simoneb) [inline package.json:41]: you'll  need to update this too. unless we know that it's axios 1.5 and up, we should try to keep it as low as possible, e.g. 1.x

**Review by simoneb — COMMENTED** [review]
Nice job, thanks for this! Just one comment about axios peerDependency

File: package.json (hunk 1 above)
Comment (ymor) [reply package.json]: Good spot! Thats now updated to >=1.0.0

Comment (simoneb) [reply package.json]: Thanks. Nope notifications for commits are not sent, so thanks for letting me know 👍

**Review by simoneb — APPROVED** [review]

**Pull request merged**